cmd/register/register_util.go

```diff
@@ -275,34 +275,76 @@ func hydrateTaskSpec(task *admin.TaskSpec, sourceCode, sourceUploadPath, version
 	return nil
 }
 
-func validateLaunchSpec(lpSpec *admin.LaunchPlanSpec) error {
-	if lpSpec == nil {
+func validateLPWithSchedule(lpSpec *admin.LaunchPlanSpec) error {
+	if lpSpec.EntityMetadata == nil || lpSpec.EntityMetadata.Schedule == nil {
 		return nil
 	}
-	if lpSpec.EntityMetadata != nil && lpSpec.EntityMetadata.Schedule != nil {
-		schedule := lpSpec.EntityMetadata.Schedule
-		var scheduleFixedParams []string
-		if lpSpec.DefaultInputs != nil {
-			for paramKey := range lpSpec.DefaultInputs.Parameters {
-				if paramKey != schedule.KickoffTimeInputArg {
-					scheduleFixedParams = append(scheduleFixedParams, paramKey)
-				}
+	schedule := lpSpec.EntityMetadata.Schedule
+	var scheduleRequiredParams []string
+	if lpSpec.DefaultInputs != nil {
+		for paramKey, paramValue := range lpSpec.DefaultInputs.Parameters {
```

Comment:
I don't think that this is right, default inputs are okay - they already have a value associated with them that can be used at execution time.

Ideally we should check the workflow spec and compare any unbound inputs (outside of KickoffTimeInputArg) and then raise that delta as the error.

Reply:
Updated @katrogan . Please review when you get a chance

Reply:
@katrogan I found this confusing.
FixedInputs *core.LiteralMap 
We have lpSpec.DefaultInputs which are basically *core.ParameterMap  and each Parameter can be 
// Types that are valid to be assigned to Behavior:
	//	*Parameter_Default
	//	*Parameter_Required
So we have lpSpec.DefaultInputs which can be of Parameter_Default or Parameter_Required.
Only if its Parameter_Default , does it have Literal value which makes sense because thats the default value for the parameter.

But if its a Parameter_Required then its doesn't have a literal value .
Can we not have required params with default values ?
The name DefaultInputs confuses me . Can you help clarify on this aspect

Reply:
we can't have required parameters with default values. they are required exactly because there is no bound value 

In the launch plan spec, anything in the DefaultInputs message can be overridden in the create execution request, or otherwise the values in the DefaultInputs block will be used. I *think* that's why we called it DefaultInputs because we default to using those values. Maybe the naming could use a refresh before flyteidl 1.0.0 :)